core/pta/hwrng.c

```diff
@@ -31,8 +31,7 @@ static TEE_Result rng_get_entropy(uint32_t types,
 	if (!e)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	for (i = 0; i < params[0].memref.size; i++)
-		e[i] = hw_get_random_byte();
+	crypto_rng_read(e, params[0].memref.size);
```

Comment:
Was thinking about this, I wasn't sure if we actually wanted to force only HW RNG with this PTA. For instance some plats (stm32mp1) seed the PRNG with a HW RNG, or add entropy from secure time sources, in that case the randomness produced by the PRNG might be "better" than otherwise. And we might want to feed that down into the REE. But maybe not..

I'll add this check here and in the plat-rcar case below.